zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/DbAuthorizationState.java

```diff
@@ -0,0 +1,97 @@
+import io.camunda.zeebe.protocol.ZbColumnFamilies;
+import io.camunda.zeebe.protocol.impl.record.value.authorization.AuthorizationRecord;
+
+public class DbAuthorizationState implements AuthorizationState, MutableAuthorizationState {
+  private final PersistedAuthorization persistedAuthorization = new PersistedAuthorization();
+
+  private final PersistedPermissions persistedPermissions = new PersistedPermissions();
+
+  private final DbLong authorizationKey;
+  private final DbString ownerKey;
+  private final DbString ownerType;
+  private final DbString resourceKey;
+  private final DbString resourceType;
+  private final DbCompositeKey<DbString, DbString> resourceCompositeKey;
+  private final DbCompositeKey<
+          DbCompositeKey<DbString, DbString>, DbCompositeKey<DbString, DbString>>
+      ownerAndResourceCompositeKey;
+  private final ColumnFamily<DbLong, PersistedAuthorization> authorizationColumnFamily;
+  private final ColumnFamily<
+          DbCompositeKey<DbCompositeKey<DbString, DbString>, DbCompositeKey<DbString, DbString>>,
+          PersistedPermissions>
+      ownerAuthorizationColumnFamily;
+
+  public DbAuthorizationState(
+      final ZeebeDb<ZbColumnFamilies> zeebeDb, final TransactionContext transactionContext) {
+    authorizationKey = new DbLong();
+    authorizationColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.AUTHORIZATIONS,
+            transactionContext,
+            authorizationKey,
+            persistedAuthorization);
+
+    ownerKey = new DbString();
+    ownerType = new DbString();
+    resourceKey = new DbString();
+    resourceType = new DbString();
+    resourceCompositeKey = new DbCompositeKey<>(resourceKey, resourceType);
+    ownerAndResourceCompositeKey =
+        new DbCompositeKey<>(new DbCompositeKey<>(ownerKey, ownerType), resourceCompositeKey);
+
+    ownerAuthorizationColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.AUTHORIZATIONS_BY_USERNAME_AND_PERMISSION,
+            transactionContext,
+            ownerAndResourceCompositeKey,
+            persistedPermissions);
+  }
+
+  @Override
+  public void createAuthorization(final AuthorizationRecord authorizationRecord) {
+    authorizationKey.wrapLong(authorizationRecord.getAuthorizationKey());
+    persistedAuthorization.setAuthorization(authorizationRecord);
+    authorizationColumnFamily.insert(authorizationKey, persistedAuthorization);
+
+    ownerKey.wrapString(authorizationRecord.getOwnerKey());
+    ownerType.wrapString(authorizationRecord.getOwnerType());
+    resourceKey.wrapString(authorizationRecord.getResourceKey());
+    resourceType.wrapString(authorizationRecord.getResourceType());
+    persistedPermissions.setPermissions(authorizationRecord.getPermissions());
+    ownerAuthorizationColumnFamily.insert(ownerAndResourceCompositeKey, persistedPermissions);
+  }
+
+  @Override
+  public PersistedPermissions getAuthorization(
```

Comment:
🔧 I feel like we should rename this to `getPermissions`. WDYT?

Reply:
I agree